processor/src/main/java/org/mapstruct/ap/services/DefaultAccessorNamingStrategy.java

```diff
@@ -0,0 +1,76 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.services;
+
+import java.beans.Introspector;
+
+import org.mapstruct.spi.AccessorNamingStrategy;
+
+/**
+ * The default JavaBeans-compliant implementation of the {@link AccessorNamingStrategy} service provider interface.
+ *
+ * @author Christian Schuster
+ */
+class DefaultAccessorNamingStrategy implements AccessorNamingStrategy {
```

Comment:
> So I'd rather keep the implementation secret and document the default behavior in the interface's Javadoc.

+1

> before returning the custom implementation.

I'm wondering when an implementor would actually use that? The implementation is fairly trivial, so I don't see much value in supporting that delegation and would rather prefer to make the exposed contract as lean as possible. Also if one works with custom name patterns in the first place, would delegation to the default implementation be useful at all?